apps/prairielearn/elements/pl-drawing/mechanicsObjects.js

```diff
@@ -1040,6 +1040,9 @@ mechanicsObjects.LatexText = fabric.util.createClass(fabric.Object, {
         if (new_text !== null) {
           this.label = new_text;
           this.gen_text(this.parse(new_text), options);
+
+          // Fire an event to ensure that the text is updated in the submission data.
+          // this.fire('modified');
```

Comment:
Isn't there something missing here?

Reply:
el oh el, I commented it out to double-check my reproduction instructions against the broken version...

```suggestion

          // Fire an event to ensure that the text is updated in the submission data.
          this.fire('modified');
```